### PR TITLE
Mixin: Correctly spell `hierarchies`

### PR DESCRIPTION
**Pull request by jacobbaungard**
Fix typo in the mixin.

Note: this is a breaking change for anyone changing this value downstream.

### CODE REVIEW

**Review by PhilipGough — APPROVED** [review]

File: jsonnet/thanos-receive-controller-mixin/config.libsonnet [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   local thanos = self,
-  // Hierarcies is a way to help mixin users to add high level hierarcies to their alerts and dashboards.
+  // Hierarchies is a way to help mixin users to add high level hierarchies to their alerts and dashboards.
   // The key in the key-value pair will be used as label name in the alerts and variable name in the dashboards.
   // The value in the key-value pair will be used as a query to fetch available values for the given label name.
-  hierarcies+:: {
+  // old spelling for backwards compatibility
```
Comment (matej-g) [inline jsonnet/thanos-receive-controller-mixin/config.libsonnet:6]: Where do we use the old spelling?

Comment (jacobbaungard) [reply jsonnet/thanos-receive-controller-mixin/config.libsonnet]: Ah, thanks! I tried to figure out a way to keep the old spelling for backwards compatibility but didn't find a good way in the end. This was a left-over comment from that attempt, removed it now.

Comment (matej-g) [reply jsonnet/thanos-receive-controller-mixin/config.libsonnet]: Got it. I think we can live with this breaking change, so should be fine 🙂.

**Review by matej-g — APPROVED** [review]
Thanks! 🦅 👁️

**Pull request merged**